rst_files/linear_algebra.rst

```diff
@@ -807,66 +704,25 @@ If so, is it always unique?
 
 The answer to both these questions is negative, as the next figure shows
 
-
-
 .. code-block:: julia
-  :class: collapse
-
-    f(x) = 0.6 * cos(4.0 * x) + 1.3
-
-    xmin, xmax = -1.0, 1.0
-    Nx = 160
-    x = range(xmin, xmax, length = Nx)
-    y = f.(x)
-    ya, yb = extrema(y)
-
-    p1 = plot(x, y, color = :black, label = [L"$f$" ""], grid = false)
-    plot!(x, ya * ones(Nx, 1), fill_between = yb * ones(Nx, 1),
-        fillalpha = 0.1, color = :blue, label = "", lw = 0)
-    plot!(zeros(2, 2), [ya ya; yb yb], lw = 3, color = :blue, label = [L"range of $f$" ""])
-    annotate!(0.04, -0.3, L"$0$", ylims = (-0.6, 3.2))
-    vline!([0], color = :black, label = "")
-    hline!([0], color = :black, label = "")
-    plot!(foreground_color_axis = :white, foreground_color_text = :white,
-        foreground_color_border = :white)
-
-    ybar = 1.5
-    plot!(x, x .* 0 .+ ybar, color = :black, linestyle = :dash, label = "")
-    annotate!(0.05, 0.8 * ybar, L"$y$")
-
-    x_vals = zeros(2, 4)
-    y_vals = similar(x_vals)
-    labels = []
-    for (i, z) ∈ enumerate([-0.35, 0.35])
-        x_vals[:, 2*i-1] = z * ones(2, 1)
-        y_vals[2, 2*i-1] = f(z)
-        labels = [labels; (z, -0.2, LaTeXString("\$x_$i\$"))]
-    end
-    plot!(x_vals, y_vals, color = :black, linestyle = :dash, label = "", annotation = labels)
-
-    p2 = plot(x, y, color = :black, label = [L"$f$" ""], grid=false)
-    plot!(x, ya*ones(Nx, 1), fill_between = yb * ones(Nx, 1),
-        fillalpha = 0.1, color = :blue, label = "", lw = 0)
-    plot!(zeros(2, 2), [ya ya; yb yb], lw = 3, color = :blue, label = [L"range of $f$" ""])
-    annotate!(0.04, -0.3, L"$0$", ylims = (-0.6, 3.2))
-    vline!([0], color = :black, label = "")
-    hline!([0], color = :black, label = "")
-    plot!(foreground_color_axis = :white, foreground_color_text = :white,
-        foreground_color_border = :white)
-
-    ybar = 2.6
-    plot!(x, x .* 0 .+ ybar, color = :black, linestyle = :dash, legend = :none)
-    annotate!(0.04, 0.91 * ybar, L"$y$")
-
-    plot(p1, p2, layout = (2, 1), size = (600, 700))
+    :class: collapse
+
+    y_min, y_max = extrema(0.6cos.(4x) + 1.3 for x in range(-2, 2, length = 100))
+    plt1 = plot(x -> 0.6cos(4x) + 1.3, xlim = (-2, 2), label = "f")
```

Comment:
We could then replace this with `plot(vals)`.

Reply:
The nice aspect of `plot` is that it takes a function and one needs not to specify a grid.

Reply:
True, and I think we should be using that frequently.

If arnav thought this was tough to read, maybe having this as a separate function would be better
```
f(x) = 0.6cos.(4x) + 1.3 f
y_min, y_max = extrema( f.(range(-2, 2, length = 100))
plt1 = plot(f, xlim = (-2, 2), label = "f")
```
Or something like that.

Reply:
After thinking about it some more, I don't think I have an issue with this. I was mainly saying that if we switched the thing above to `vals`, then we might as well switch this. But I like the function feature.